services/shhext/api.go

```diff
@@ -186,7 +188,42 @@ func (api *PublicAPI) GetNewFilterMessages(filterID string) ([]*whisper.Message,
 	if err != nil {
 		return nil, err
 	}
-	return api.service.deduplicator.Deduplicate(msgs), err
+
+	dedupMessages := api.service.deduplicator.Deduplicate(msgs)
+
+	// Attempt to decrypt message, otherwise leave unchanged
+	for _, msg := range dedupMessages {
+
+		// Msg.Dst is empty is a public message, nothing to do
+		if msg.Dst != nil {
+			// There's probably a better way to do this
+			keyBytes, err := hexutil.Bytes(msg.Dst).MarshalText()
+			if err != nil {
+				return nil, err
+			}
+
+			privateKey, err := api.service.w.GetPrivateKey(string(keyBytes))
+			if err != nil {
+				return nil, err
+			}
+
+			// This needs to be pushed down in the protocol message
+			publicKey, err := crypto.UnmarshalPubkey(msg.Sig)
```

Comment:
For now we rely on whisper to ensure authentication of the message (whisper verifies the signature), eventually we want to move the authentication layer in the encryption layer, but for now this step relies on whisper/transport.